src/lib/transaction_database/transaction_database.ml

```diff
@@ -45,14 +54,61 @@ let create ?directory_name () =
 
 let close {database; _} = Database.close database
 
-let add t (public_key : Public_key.Compressed.t) transaction =
+let add t transaction =
   match Hash_set.strict_add t.cache.all_transactions transaction with
   | Error _ ->
       ()
   | Ok () ->
-      Database.set t.database ~key:transaction ~data:public_key ;
-      Hashtbl.add_multi t.cache.user_transactions ~key:public_key
-        ~data:transaction
+      let sender_and_receiver_pairs =
+        match transaction with
+        | Fee_transfer (One (pk, _)) ->
+            [Payment_participants.{receiver= Some pk; sender= None}]
+        | Fee_transfer (Two ((pk1, _), (pk2, _))) ->
```

Comment:
We always have a receiver right? So maybe just have pk instead of pk option as the receiver type?